packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx

```diff
@@ -466,6 +467,7 @@ export const useGridColumnHeaders = (props: UseGridColumnHeadersProps) => {
           role="row"
           aria-rowindex={depth + 1}
           ownerState={rootProps}
+          style={{ height: groupHeaderHeight }}
```

Comment:
Not sure whether this could be classified as a breaking change or not. The height of the headers are applied to the header row rather than header cells so that it does not collapse when there are no cells in it.